buildingmotif/api/views/model.py

```diff
@@ -200,9 +204,17 @@ def validate_model(models_id: int) -> flask.Response:
                 "message": f"Libraries with ids {nonexistent_libraries} do not exist"
             }, status.HTTP_400_BAD_REQUEST
 
-    # if shape_collections is empty, model.validate will default
-    # to the model's manifest
-    vaildation_context = model.validate(shape_collections)
+    # temporarily change the SHACL engine
+    bm = get_building_motif()
+    old_shacl_engine = bm.shacl_engine
+    bm.shacl_engine = shacl_engine
```

Comment:
instead of swapping out the global engine, would it make sense to expose the engine as a parameter like in the `shacl_validate` method `Model.validate` is calling?

Reply:
I've done this for the methods I could think of in the code base. If the shacl_engine parameter is not provided in those methods, then it will default to whatever BuildingMOTIF is currently using